<div style=" background-color: RGB(0,114,200);" >
<h1 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">PROJET 4 DATA ANALYST</h1>
<h2 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">Réalisez une étude de santé publique avec R ou Python
</h2>
</div>

# OBJECTIF DE CE NOTEBOOK

Bienvenue dans l'outil plébiscité par les analystes de données Jupyter.

Il s'agit d'un outil permettant de mixer et d'alterner codes, textes et graphique.

Cet outil est formidable pour plusieurs raisons:

+ il permet de tester des lignes de codes au fur et à mesure de votre rédaction, de constater immédiatement le résultat d'un instruction, de la corriger si nécessaire.
+ De rédiger du texte pour expliquer l'approche suivie ou les résultats d'une analyse et de le mettre en forme grâce à du code html ou plus simple avec **Markdown**
+ d'agrémenter de graphiques

Pour vous aider dans vos premiers pas à l'usage de Jupyter et de Python, nous avons rédigé ce notebook en vous indiquant les instructions à suivre.

Il vous suffit pour cela de saisir le code Python répondant à l'instruction donnée.

Vous verrez de temps à autre le code Python répondant à une instruction donnée mais cela est fait pour vous aider à comprendre la nature du travail qui vous est demandée.

Et garder à l'esprit, qu'il n'y a pas de solution unique pour résoudre un problème et qu'il y a autant de résolutions de problèmes que de développeurs ;)...



Note jeremy 
Est ce qu'il faut faire le calcul de la sous nutrition sur les pays qu'on a ? 
Est ce qu'il faut faire des graphiques ?
Rajouter le soja
La liste des céréales est difficile a trouver ...

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 1 - Importation des librairies et chargement des fichiers</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.1 - Importation des librairies</h3>
</div>

In [224]:
#Importation de la librairie Pandas
import pandas as pd

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.2 - Chargement des fichiers Excel</h3>
</div>

In [225]:
#Importation du fichier population.csv
population = pd.read_csv('population.csv')

#Importation du fichier dispo_alimentaire.csv
dispoAlimentaire = pd.read_csv('dispo_alimentaire.csv')

#Importation du fichier aide_alimentaire.csv
aideAlimentaire = pd.read_csv('aide_alimentaire.csv')

#Importation du fichier sous_nutrition.csv
sousNutrition = pd.read_csv('sous_nutrition.csv')

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 2 - Analyse exploratoire des fichiers</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.1 - Analyse exploratoire du fichier population</h3>
</div>

In [226]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(population.shape[0]))
print("Le tableau comporte {} colonne(s)".format(population.shape[1]))

Le tableau comporte 1416 observation(s) ou article(s)
Le tableau comporte 3 colonne(s)


In [227]:
#Consulter le nombre de colonnes
print("Le tableau comporte {} colonne(s)".format(len(population.columns)))

#La nature des données dans chacune des colonnes
# Convertir la série des types de données en un DataFrame
dtypes_df = population.dtypes.to_frame()
# Formater le DataFrame en une chaîne
dtypes_str = dtypes_df.to_string()
# Afficher le résultat
print("La nature des colonnes est :\n{}".format(dtypes_str))

#Le nombre de valeurs présentes dans chacune des colonnes
print("Le nombre de valeurs présentes dans chacune des colonnes est :\n{}".format(population.count()))

Le tableau comporte 3 colonne(s)
La nature des colonnes est :
              0
Zone     object
Année     int64
Valeur  float64
Le nombre de valeurs présentes dans chacune des colonnes est :
Zone      1416
Année     1416
Valeur    1416
dtype: int64


Le tableau comporte 3 colonne(s)

La nature des colonnes est :
Zone     object
Année     int64
Valeur  float64

Le nombre de valeurs présentes dans chacune des colonnes est :
Zone      1416
Année     1416
Valeur    1416

In [228]:
#Affichage les 5 premières lignes de la table
print("Les 5 premières lignes de la table population")
print(population.head())


Les 5 premières lignes de la table population
          Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113


Les 5 premières lignes de la table population
          Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113

In [229]:
#Nous allons harmoniser les unités. Pour cela, nous avons décidé de multiplier la population par 1000
#Multiplication de la colonne valeur par 1000

print("Nous allons harmoniser les unités en multipliant la population par 1000.")

# Multiplication de la colonne "Valeur" par 1000
population['Population'] = population['Valeur'] * 1000

# Convertir les valeurs en entiers
population['Population'] = population['Population'].astype(int)

# Afficher le résultat
print("Nous avons harmonisé les unités en multipliant la population par 1000 :\n{}".format(population))


Nous allons harmoniser les unités en multipliant la population par 1000.
Nous avons harmonisé les unités en multipliant la population par 1000 :
             Zone  Année     Valeur  Population
0     Afghanistan   2013  32269.589    32269589
1     Afghanistan   2014  33370.794    33370794
2     Afghanistan   2015  34413.603    34413603
3     Afghanistan   2016  35383.032    35383032
4     Afghanistan   2017  36296.113    36296113
...           ...    ...        ...         ...
1411     Zimbabwe   2014  13586.707    13586707
1412     Zimbabwe   2015  13814.629    13814629
1413     Zimbabwe   2016  14030.331    14030331
1414     Zimbabwe   2017  14236.595    14236595
1415     Zimbabwe   2018  14438.802    14438802

[1416 rows x 4 columns]


Nous avons harmonisé les unités en multipliant la population par 1000 :
0       32269589
1       33370794
2       34413603
3       35383032
4       36296113
          ...   
1411    13586707
1412    13814629
1413    14030331
1414    14236595
1415    14438802

In [230]:
# Supprimer la colonne "Valeur" originale
population.drop(columns=['Valeur'], inplace=True)

# Afficher le résultat
print(population.head())

          Zone  Année  Population
0  Afghanistan   2013    32269589
1  Afghanistan   2014    33370794
2  Afghanistan   2015    34413603
3  Afghanistan   2016    35383032
4  Afghanistan   2017    36296113


   Zone  Année  Population
0  Afghanistan   2013   32269.589
1  Afghanistan   2014   33370.794
2  Afghanistan   2015   34413.603
3  Afghanistan   2016   35383.032
4  Afghanistan   2017   36296.113

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.2 - Analyse exploratoire du fichier disponibilité alimentaire</h3>
</div>

In [231]:
#Afficher les dimensions du dataset

print("Le tableau comporte {} observation(s) ou article(s)".format(dispoAlimentaire.shape[0]))

Le tableau comporte 15605 observation(s) ou article(s)


Le tableau comporte 15605 observation(s) ou article(s)

In [232]:
#Consulter le nombre de colonnes
print("Le tableau comporte {} colonne(s)".format(dispoAlimentaire.shape[1]))


Le tableau comporte 18 colonne(s)


Le tableau comporte 18 colonne(s)

In [233]:
#Affichage les 5 premières lignes de la table
print("Les 5 premières lignes de la table dispositions alimentaires sont :")
print(dispoAlimentaire.head())


Les 5 premières lignes de la table dispositions alimentaires sont :
          Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (

Les 5 premières lignes de la table dispositions alimentaires sont :
          Zone                Produit   Origine  Aliments pour animaux  Autres Utilisations  ...  Pertes  Production  Semences  Traitement  Variation de stock
0  Afghanistan       Abats Comestible   animale                    NaN                  NaN  ...     NaN        53.0       NaN         NaN                 NaN
1  Afghanistan        Agrumes, Autres  vegetale                    NaN                  NaN  ...     2.0         3.0       NaN         NaN                 NaN
2  Afghanistan  Aliments pour enfants  vegetale                    NaN                  NaN  ...     NaN         NaN       NaN         NaN                 NaN
3  Afghanistan                 Ananas  vegetale                    NaN                  NaN  ...     NaN         NaN       NaN         NaN                 NaN
4  Afghanistan                Bananes  vegetale                    NaN                  NaN  ...     NaN         NaN       NaN         NaN                 NaN

In [234]:
#remplacement des NaN dans le dataset par des 0
dispoAlimentaire = dispoAlimentaire.fillna(0)
print(dispoAlimentaire.head())


          Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1

     Zone                Produit   Origine  Aliments pour animaux  Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  Disponibilité alimentaire en quantité (kg/personne/an)  ...  Importations - Quantité  Nourriture  Pertes  Production  Semences  Traitement  Variation de stock
0  Afghanistan       Abats Comestible   animale                    0.0                  0.0                                             5.0                                               1.72       ...                      0.0        53.0     0.0        53.0       0.0         0.0                 0.0
1  Afghanistan        Agrumes, Autres  vegetale                    0.0                  0.0                                             1.0                                               1.29       ...                     40.0        39.0     2.0         3.0       0.0         0.0                 0.0
2  Afghanistan  Aliments pour enfants  vegetale                    0.0                  0.0                                             1.0                                               0.06       ...                      2.0         2.0     0.0         0.0       0.0         0.0                 0.0
3  Afghanistan                 Ananas  vegetale                    0.0                  0.0                                             0.0                                               0.00       ...                      0.0         0.0     0.0         0.0       0.0         0.0                 0.0
4  Afghanistan                Bananes  vegetale                    0.0                  0.0                                             4.0                                               2.70       ...                     82.0        82.0     0.0         0.0       0.0         0.0                 0.0

In [235]:
#multiplication de toutes les lignes contenant des milliers de tonnes en Kg
colonnes_a_convertir = ['Autres Utilisations', 'Disponibilité intérieure', 'Exportations - Quantité',
                        'Importations - Quantité', 'Nourriture', 'Pertes', 'Production', 'Semences', 
                        'Traitement', 'Variation de stock']

for colonne in colonnes_a_convertir:
    dispoAlimentaire[colonne] *= 1000  # Convertir de milliers de tonnes à kg



In [236]:
#Affichage les 5 premières lignes de la table
print(dispoAlimentaire.head())

          Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1

 Aliments pour animaux  Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  Disponibilité alimentaire en quantité (kg/personne/an)  Disponibilité de matière grasse en quantité (g/personne/jour)  Disponibilité de protéines en quantité (g/personne/jour)  ...  Nourriture   Pertes  Production  Semences  Traitement  Variation de stock
0                      0                    0                                         5000000                                            1720000                                                  200000                                                         770000         ...    53000000        0    53000000         0           0                   0
1                      0                    0                                         1000000                                            1290000                                                   10000                                                          20000         ...    39000000  2000000     3000000         0           0                   0
2                      0                    0                                         1000000                                              60000                                                   10000                                                          30000         ...     2000000        0           0         0           0                   0
3                      0                    0                                               0                                                  0                                                       0                                                              0         ...           0        0           0         0           0                   0
4                      0                    0                                         4000000                                            2700000                                                   20000                                                          50000         ...    82000000        0           0         0           0                   0

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - Analyse exploratoire du fichier aide alimentaire</h3>
</div>

In [237]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(aideAlimentaire.shape[0]))
print("Le tableau comporte {} colonne(s)".format(aideAlimentaire.shape[1]))

Le tableau comporte 1475 observation(s) ou article(s)
Le tableau comporte 4 colonne(s)


In [238]:
#Consulter le nombre de colonnes
print("Le nombre de colonnes est : {}".format(aideAlimentaire.shape[1]))


Le nombre de colonnes est : 4


In [239]:
#Affichage les 5 premières lignes de la table
print("les 5 premières lignes de la table aideAlimentaire sont : ")
print(aideAlimentaire.head())

les 5 premières lignes de la table aideAlimentaire sont : 
  Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504


In [240]:
#changement du nom de la colonne Pays bénéficiaire par Zone
aideAlimentaire.rename(columns={'Pays bénéficiaire':'Zone'}, inplace=True)
print(aideAlimentaire.head())


          Zone  Année              Produit  Valeur
0  Afghanistan   2013  Autres non-céréales     682
1  Afghanistan   2014  Autres non-céréales     335
2  Afghanistan   2013         Blé et Farin   39224
3  Afghanistan   2014         Blé et Farin   15160
4  Afghanistan   2013             Céréales   40504


In [241]:
#Multiplication de la colonne Aide_alimentaire qui contient des tonnes par 1000 pour avoir des kg
aideAlimentaire['Valeur'] = aideAlimentaire['Valeur']*1000


In [242]:
#Affichage les 5 premières lignes de la table
print(aideAlimentaire.head())

          Zone  Année              Produit    Valeur
0  Afghanistan   2013  Autres non-céréales    682000
1  Afghanistan   2014  Autres non-céréales    335000
2  Afghanistan   2013         Blé et Farin  39224000
3  Afghanistan   2014         Blé et Farin  15160000
4  Afghanistan   2013             Céréales  40504000


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - Analyse exploratoire du fichier sous nutrition</h3>
</div>

In [243]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(sousNutrition.shape[0]))


Le tableau comporte 1218 observation(s) ou article(s)


In [244]:
#Consulter le nombre de colonnes
print("Le tableau comporte {} colonne(s)".format(sousNutrition.shape[1]))

Le tableau comporte 3 colonne(s)


In [245]:
#Afficher les 5 premières lignes de la table
print(sousNutrition.head())


          Zone      Année Valeur
0  Afghanistan  2012-2014    8.6
1  Afghanistan  2013-2015    8.8
2  Afghanistan  2014-2016    8.9
3  Afghanistan  2015-2017    9.7
4  Afghanistan  2016-2018   10.5


In [246]:
#Conversion de la colonne sous nutrition en numérique
sousNutrition['Valeur'] = pd.to_numeric(sousNutrition['Valeur'], errors='coerce')



In [247]:
#Conversion de la colonne (avec l'argument errors=coerce qui permet de convertir automatiquement les lignes qui ne sont pas des nombres en NaN)
#Puis remplacement des NaN en 0
sousNutrition['Valeur'] = sousNutrition['Valeur'].fillna(0)
print(sousNutrition['Valeur'])


0        8.6
1        8.8
2        8.9
3        9.7
4       10.5
        ... 
1213     0.0
1214     0.0
1215     0.0
1216     0.0
1217     0.0
Name: Valeur, Length: 1218, dtype: float64


In [248]:
#changement du nom de la colonne Valeur par sous_nutrition
sousNutrition = sousNutrition.rename(columns={'Valeur': 'sous_nutrition'})
print(sousNutrition)

             Zone      Année  sous_nutrition
0     Afghanistan  2012-2014             8.6
1     Afghanistan  2013-2015             8.8
2     Afghanistan  2014-2016             8.9
3     Afghanistan  2015-2017             9.7
4     Afghanistan  2016-2018            10.5
...           ...        ...             ...
1213     Zimbabwe  2013-2015             0.0
1214     Zimbabwe  2014-2016             0.0
1215     Zimbabwe  2015-2017             0.0
1216     Zimbabwe  2016-2018             0.0
1217     Zimbabwe  2017-2019             0.0

[1218 rows x 3 columns]


In [249]:
#Multiplication de la colonne sous_nutrition par 1000000
sousNutrition['sous_nutrition'] *= 1000000

# Conversion en chiffres entiers
sousNutrition['sous_nutrition'] = sousNutrition['sous_nutrition'].astype(int)






In [250]:
#Afficher les 5 premières lignes de la table
print(sousNutrition.head())

          Zone      Année  sous_nutrition
0  Afghanistan  2012-2014         8600000
1  Afghanistan  2013-2015         8800000
2  Afghanistan  2014-2016         8900000
3  Afghanistan  2015-2017         9700000
4  Afghanistan  2016-2018        10500000


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.1 - Proportion de personnes en sous nutrition</h3>
</div>

In [251]:
# TRAVAIL SUR LA TABLE SOUS NUTRITION


# Déploiement du groupe d'année sur une ligne ['2012-2014'] => ['2012', '2013', '2014']
sousNutrition['Année'] = sousNutrition['Année'].str.split('-').apply(lambda x: list(range(int(x[0]), int(x[1])+1)))
sousNutrition = sousNutrition.explode('Année')
sousNutrition['Année'] = sousNutrition['Année'].astype(int)

# Réinitialiser les index /garantit que chaque ligne a un index unique et que les index sont contigus, ce qui facilite l'accès et la manipulation des données.
sousNutrition = sousNutrition.reset_index(drop=True)

# Convertir la colonne 'Année' en type entier (int)
sousNutrition['Année'] = sousNutrition['Année'].astype(int)


# ************************************************************


# Jointure entre les tables population et sousNutrition sur les colonnes 'Zone' et 'Année'
jointure = pd.merge(population, sousNutrition, on=['Zone', 'Année'])

# Filtrer les résultats pour ne conserver que les lignes avec l'année 2017
resultats_2017 = jointure[jointure['Année'] == 2017]

#Regrouper les données en fonction de la colonne Zone 
#puis retenir la première valeur de la colonne Population et la dernière valeurs de la colonne sous_nutrition
# les données de la FAO en sous-nutrition sont déjà une moyenne sur trois ans (vérification sur le site FAOSTAT), donc je récupère la valeur la plus récente
# Trouver la valeur de personnes en sous-nutrition pour chaque pays en 2017
resultats_2017 = resultats_2017.groupby('Zone').agg({'Population': 'first', 'sous_nutrition': 'last'}).reset_index()

# Convertir la colonne 'sous_nutrition' en type entier
resultats_2017['sous_nutrition'] = resultats_2017['sous_nutrition'].astype(int)

# Calculer la proportion de sous-nutrition
resultats_2017['proportion_sous_nutrition'] = resultats_2017['sous_nutrition'] / resultats_2017['Population']*100

# Arrondir les valeurs à deux chiffres après la virgule et rajouter le signe % à la fin
resultats_2017['proportion_sous_nutrition'] = resultats_2017['proportion_sous_nutrition'].round(2).astype(str) + '%'



In [252]:
#Affichage du dataset
# Afficher le DataFrame résultant
print(resultats_2017)


                      Zone  Population  sous_nutrition  \
0              Afghanistan    36296113        11100000   
1           Afrique du Sud    57009756         3300000   
2                  Albanie     2884169          100000   
3                  Algérie    41389189         1200000   
4                Allemagne    82658409               0   
..                     ...         ...             ...   
198  États-Unis d'Amérique   325084756               0   
199               Éthiopie   106399924        21500000   
200              Îles Cook       17507               0   
201          Îles Marshall       58058               0   
202           Îles Salomon      636039               0   

    proportion_sous_nutrition  
0                      30.58%  
1                       5.79%  
2                       3.47%  
3                        2.9%  
4                        0.0%  
..                        ...  
198                      0.0%  
199                    20.21%  
200            

In [253]:

# Calcul et affichage du nombre de personnes en état de sous-nutrition en 2017 avec des séparateurs entre les milliers
total_sous_nutrition = round(resultats_2017['sous_nutrition'].sum())

# Utiliser la méthode format avec le spécificateur de format {:,} pour ajouter des séparateurs entre les milliers
formatted_total_sous_nutrition = "{:,.0f}".format(total_sous_nutrition)

print("Le nombre de personnes en état de sous-nutrition en 2017 est de : {} personnes".format(formatted_total_sous_nutrition))

#calcul du nombre de personnes dans la population mondiale en 2017
total_population = round(resultats_2017['Population'].sum())
formatted_total_population = "{:,.0f}".format(total_population)
print("Le nombre de personnes dans la population mondiale en 2017 est de : {} personnes".format(formatted_total_population))

# Calcul et affichage du pourcentage de personnes en état de sous-nutrition dans le monde en 2017
pourcentage_sous_nutrition = total_sous_nutrition / total_population * 100
pourcentage_sous_nutrition_arrondi = round(pourcentage_sous_nutrition, 2)
print("Le pourcentage de personnes en état de sous-nutrition dans le monde en 2017 est de : {}%".format(pourcentage_sous_nutrition_arrondi))


Le nombre de personnes en état de sous-nutrition en 2017 est de : 544,200,000 personnes
Le nombre de personnes dans la population mondiale en 2017 est de : 7,543,798,769 personnes
Le pourcentage de personnes en état de sous-nutrition dans le monde en 2017 est de : 7.21%


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.2 - Nombre théorique de personnes qui pourraient être nourries</h3>
</div>

In [254]:
#Combien mange en moyenne un être humain ? Source => dispoAlimentaire
# Calculer la disponibilité alimentaire moyenne par personne par produit

disponibilite_moyenne_par_personne = dispoAlimentaire[['Produit', 'Disponibilité alimentaire en quantité (kg/personne/an)']].groupby('Produit').mean()
print(disponibilite_moyenne_par_personne)

                        Disponibilité alimentaire en quantité (kg/personne/an)
Produit                                                                       
Abats Comestible                                                 2.730345     
Agrumes, Autres                                                  1.306688     
Alcool, non Comestible                                           0.000000     
Aliments pour enfants                                            0.380936     
Ananas                                                           3.890536     
...                                                                   ...     
Viande de Suides                                                14.407616     
Viande de Volailles                                             20.561322     
Viande, Autre                                                    1.800977     
Vin                                                              6.161337     
Épices, Autres                                      

In [255]:
#Filtrer la table population sur l'année 2017 
population_2017 = population[population['Année'] == 2017]

#Faire une jointure entre les tables disponibilité alimentaire et population sur la colonne Zone
dispoAlimentaire = pd.merge(dispoAlimentaire, population_2017, on='Zone')



In [256]:
#Affichage du nouveau dataframe
# Afficher les premières lignes du DataFrame
print(dispoAlimentaire.head())

          Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1

In [257]:
# Calculer la somme de la disponibilité alimentaire en calories par personne et par jour par pays
Nbre_cal_journaliere_par_personne_disponibles = dispoAlimentaire.groupby('Zone')['Disponibilité alimentaire (Kcal/personne/jour)'].sum().reset_index()
Nbre_cal_journaliere_par_personne_disponibles.rename(columns={'Disponibilité alimentaire (Kcal/personne/jour)': 'Nbre_cal_journaliere_par_personne_disponibles'}, inplace=True)

# Fusionner les DataFrames sur la colonne 'Zone'
dispoAlimentaire = pd.merge(dispoAlimentaire, Nbre_cal_journaliere_par_personne_disponibles, on='Zone')

# Grouper par pays et obtenir la première ligne de chaque groupe
NourritureDispo = dispoAlimentaire.groupby('Zone').first().reset_index()[['Zone', 'Population', 'Nbre_cal_journaliere_par_personne_disponibles']]

# Calculer la colonne Nbre_cal_dispo_totale
NourritureDispo['Nbre_cal_dispo_totale'] = NourritureDispo['Population'] * NourritureDispo['Nbre_cal_journaliere_par_personne_disponibles']

# Afficher les premières lignes du DataFrame mis à jour
print(NourritureDispo.head())



             Zone  Population  Nbre_cal_journaliere_par_personne_disponibles  \
0     Afghanistan    36296113                                         2087.0   
1  Afrique du Sud    57009756                                         3020.0   
2         Albanie     2884169                                         3188.0   
3         Algérie    41389189                                         3293.0   
4       Allemagne    82658409                                         3503.0   

   Nbre_cal_dispo_totale  
0           7.574999e+10  
1           1.721695e+11  
2           9.194731e+09  
3           1.362946e+11  
4           2.895524e+11  


In [258]:

# Nombre moyen de calories nécessaires par personne par jour (valeur fictive)
calories_necessaires_par_personne = 2350  # en calories

# Calculer le nombre d'humains pouvant être nourris
NourritureDispo['Nb_humains_nourris'] = NourritureDispo['Nbre_cal_dispo_totale'] / calories_necessaires_par_personne

# Calculer le nombre total d'humains nourris
nombre_total_humains_nourris = NourritureDispo['Nb_humains_nourris'].sum()

print("Le nombre total d'humains pouvant être nourris est d'environ : {:.2f} milliards d'humains".format(nombre_total_humains_nourris / 1e9))




Le nombre total d'humains pouvant être nourris est d'environ : 8.90 milliards d'humains


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.3 - Nombre théorique de personne qui pourrait être nourrie avec les produits végétaux</h3>
</div>

In [259]:
#Transfert des données avec les végétaux dans un nouveau dataframe
dispoAlimentaire_vegetaux = dispoAlimentaire[dispoAlimentaire['Origine'] == 'vegetale']
print(dispoAlimentaire_vegetaux.head())

          Zone                Produit   Origine  Aliments pour animaux  \
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   
6  Afghanistan                  Bière  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   
6                  0.0                                             0.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
1                                               1

In [260]:
#Calcul du nombre de kcal disponible pour les végétaux

# Supprimer la colonne 'Nbre_cal_journaliere_par_personne_disponibles' du DataFrame dispoAlimentaire_vegetaux car elle regroupe toutes les origines
dispoAlimentaire_vegetaux = dispoAlimentaire_vegetaux.drop('Nbre_cal_journaliere_par_personne_disponibles', axis=1)

#calculer le Nbre_cal_vege_journaliere_par_personne_disponibles pour les produits végétaux
Nbre_cal_vege_journaliere_par_personne_disponibles = dispoAlimentaire_vegetaux.groupby('Zone')['Disponibilité alimentaire (Kcal/personne/jour)'].sum().reset_index()
Nbre_cal_vege_journaliere_par_personne_disponibles.rename(columns={'Disponibilité alimentaire (Kcal/personne/jour)': 'Nbre_cal_vege_journaliere_par_personne_disponibles'}, inplace=True)

#Ajouter cette colonne au DataFrame dispoAlimentaire_vegetaux
dispoAlimentaire_vegetaux = pd.merge(dispoAlimentaire_vegetaux, Nbre_cal_vege_journaliere_par_personne_disponibles, on='Zone')

#Grouper par pays et obtenir la première ligne de chaque groupe
NourritureDispo_vegetaux = dispoAlimentaire_vegetaux.groupby('Zone').first().reset_index()[['Zone', 'Population', 'Nbre_cal_vege_journaliere_par_personne_disponibles']]
print(NourritureDispo_vegetaux.head())

             Zone  Population  \
0     Afghanistan    36296113   
1  Afrique du Sud    57009756   
2         Albanie     2884169   
3         Algérie    41389189   
4       Allemagne    82658409   

   Nbre_cal_vege_journaliere_par_personne_disponibles  
0                                             1871.0   
1                                             2533.0   
2                                             2203.0   
3                                             2915.0   
4                                             2461.0   


In [261]:
#Calculer la colonne Nbre_cal_vege_dispo_totale
NourritureDispo_vegetaux['Nbre_cal_vege_dispo_totale'] = NourritureDispo_vegetaux['Population'] * NourritureDispo_vegetaux['Nbre_cal_vege_journaliere_par_personne_disponibles']

#Calculer le nombre d'humains pouvant être nourris avec les produits végétaux
NourritureDispo_vegetaux['Nb_humains_nourris_vegetaux'] = NourritureDispo_vegetaux['Nbre_cal_vege_dispo_totale'] / calories_necessaires_par_personne

#Calculer le nombre total d'humains nourris avec les produits végétaux
nombre_total_humains_nourris_vegetaux = NourritureDispo_vegetaux['Nb_humains_nourris_vegetaux'].sum()
print("Le nombre total d'humains pouvant être nourris avec les produits végétaux est d'environ : {:.2f} milliards d'humains".format(nombre_total_humains_nourris_vegetaux / 1e9))

Le nombre total d'humains pouvant être nourris avec les produits végétaux est d'environ : 7.35 milliards d'humains


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.4 - Utilisation de la disponibilité intérieure</h3>
</div>

In [262]:
#Calcul de la disponibilité totale
dispo_int = dispoAlimentaire['Disponibilité intérieure'].sum()
print("La disponibilité intérieure mondiale est de : {:.2f} kilos".format(dispo_int))
print("La disponibilité intérieure mondiale est de : {:.2f} tonnes".format(dispo_int/1000))


La disponibilité intérieure mondiale est de : 9733927000.00 kilos
La disponibilité intérieure mondiale est de : 9733927.00 tonnes


In [263]:
#création d'une boucle for pour afficher les différentes valeurs en fonction des colonnes aliments pour animaux, pertes, nourritures, 
colonnes = ['Aliments pour animaux', 'Pertes', 'Nourriture']
for colonne in colonnes:
    dispo_int_colonne = dispoAlimentaire[colonne].sum()
    print("La disponibilité intérieure mondiale en {} est de : {:.2f} kilos".format(colonne, dispo_int_colonne))


La disponibilité intérieure mondiale en Aliments pour animaux est de : 1288002.00 kilos
La disponibilité intérieure mondiale en Pertes est de : 452283000.00 kilos
La disponibilité intérieure mondiale en Nourriture est de : 4805525000.00 kilos


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.5 - Utilisation des céréales</h3>
</div>

In [264]:
#Création d'une liste avec toutes les variables du fichier DispoAlimentaire
variables = ['Autres Utilisations', 
             'Aliments pour animaux',
             'Nourriture', 
             'Pertes', 
             'Semences', 
             'Traitement',
             'Exportations - Quantité',
             'Importations - Quantité',
             'Production',
             'Variation de stock',
             'Disponibilité intérieure'
             ]


In [265]:
#création d'un dataframe qui filtre sur toutes les céréales ['Blé', 'Riz (Eq Blanchi)', 'Orge', 'Maïs', 'Millet', 'Seigle', 'Avoine', 'Sorgho', 'Céréales, Autres']
cereales = dispoAlimentaire[dispoAlimentaire['Produit'].isin(['Blé', 'Riz (Eq Blanchi)', 'Orge', 'Maïs', 'Millet', 'Seigle', 'Avoine', 'Sorgho', 'Céréales, Autres'])]


In [266]:
# Calculer la proportion totale de 'Aliments pour animaux' par rapport à la disponibilité intérieure totale
proportion_aliments_animaux = cereales['Aliments pour animaux'].sum() / cereales['Disponibilité intérieure'].sum() * 100
print("La proportion totale d'Aliments pour animaux par rapport à la disponibilité intérieure totale est de : {:.2f}%".format(proportion_aliments_animaux))



La proportion totale d'Aliments pour animaux par rapport à la disponibilité intérieure totale est de : 0.04%


In [267]:

# Calculer la proportion totale de 'Nourriture'humaine par rapport à la disponibilité intérieure totale
proportion_nourriture = cereales['Nourriture'].sum() / cereales['Disponibilité intérieure'].sum() * 100
print("La proportion totale de Nourriture humaine par rapport à la disponibilité intérieure totale est de : {:.2f}%".format(proportion_nourriture))

La proportion totale de Nourriture humaine par rapport à la disponibilité intérieure totale est de : 42.91%


In [268]:

# Calculer la proportion totale par variable par rapport à la disponibilité intérieure totale
proportions = cereales[variables].sum() / cereales['Disponibilité intérieure'].sum() * 100
print(proportions)
print("La différence entre les (importations et les exportations) + la variation de stock  annule le surplus de production de céréales")

Autres Utilisations           9.782830
Aliments pour animaux         0.036143
Nourriture                   42.906006
Pertes                        4.486516
Semences                      2.847285
Traitement                    3.858986
Exportations - Quantité      17.636483
Importations - Quantité      16.461814
Production                  104.934974
Variation de stock           -3.760053
Disponibilité intérieure    100.000000
dtype: float64
La différence entre les (importations et les exportations) + la variation de stock  annule le surplus de production de céréales


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.6 - Pays avec la proportion de personnes sous-alimentée la plus forte en 2017</h3>
</div>

In [269]:
#Création de la colonne proportion par pays

resultats_2017['proportion_sous_nutrition'] = (resultats_2017['sous_nutrition'] / resultats_2017['Population']) * 100
print(resultats_2017.head())

             Zone  Population  sous_nutrition  proportion_sous_nutrition
0     Afghanistan    36296113        11100000                  30.581787
1  Afrique du Sud    57009756         3300000                   5.788483
2         Albanie     2884169          100000                   3.467203
3         Algérie    41389189         1200000                   2.899308
4       Allemagne    82658409               0                   0.000000


In [270]:
#affichage après tri des 10 pires pays
pays_max_sous_nutrition = resultats_2017.nlargest(10, 'proportion_sous_nutrition')
print(pays_max_sous_nutrition)
(sort value)

SyntaxError: invalid syntax. Perhaps you forgot a comma? (3651830103.py, line 4)

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.7 - Pays qui ont le plus bénéficié d'aide alimentaire depuis 2013</h3>
</div>

In [ ]:
#calcul du total de l'aide alimentaire par pays
# Afficher les années disponibles dans la table aideAlimentaire
print("Les années disponibles dans la table Aide Alimentaires sont :", aideAlimentaire['Année'].unique())

#calcul du total de l'aide alimentaire par pays
aideAlimentaire['Total_aide_alimentaire'] = aideAlimentaire.groupby('Zone')['Valeur'].transform('sum')
print("Le total des aides alimentaires par pays depuis 2013 est :\n{}".format(aideAlimentaire))



Les années disponibles dans la table Aide Alimentaires sont : [2013 2014 2015 2016]
Le total des aides alimentaires par pays depuis 2013 est :
             Zone  Année                   Produit    Valeur  \
0     Afghanistan   2013       Autres non-céréales    682000   
1     Afghanistan   2014       Autres non-céréales    335000   
2     Afghanistan   2013              Blé et Farin  39224000   
3     Afghanistan   2014              Blé et Farin  15160000   
4     Afghanistan   2013                  Céréales  40504000   
...           ...    ...                       ...       ...   
1470     Zimbabwe   2015  Mélanges et préparations     96000   
1471     Zimbabwe   2013              Non-céréales   5022000   
1472     Zimbabwe   2014              Non-céréales   2310000   
1473     Zimbabwe   2015              Non-céréales    306000   
1474     Zimbabwe   2013                Riz, total     64000   

      Total_aide_alimentaire  
0                  185452000  
1                  1854520

In [ ]:
#affichage après tri des 10 pays qui ont bénéficié le plus de l'aide alimentaire
pays_max_aide_alimentaire = aideAlimentaire.nlargest(10, 'Total_aide_alimentaire')
print(pays_max_aide_alimentaire)

                           Zone  Année              Produit     Valeur  \
1002  République arabe syrienne   2013  Autres non-céréales     559000   
1003  République arabe syrienne   2014  Autres non-céréales      30000   
1004  République arabe syrienne   2015  Autres non-céréales    3196000   
1005  République arabe syrienne   2013         Blé et Farin  122708000   
1006  République arabe syrienne   2014         Blé et Farin  103823000   
1007  République arabe syrienne   2015         Blé et Farin   68726000   
1008  République arabe syrienne   2016         Blé et Farin   11428000   
1009  République arabe syrienne   2013         BulgurFarEnt   12033000   
1010  République arabe syrienne   2014         BulgurFarEnt   47816000   
1011  République arabe syrienne   2015         BulgurFarEnt   24673000   

      Total_aide_alimentaire  
1002              1858943000  
1003              1858943000  
1004              1858943000  
1005              1858943000  
1006              1858943000  

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.8 - Evolution des 5 pays qui ont le plus bénéficiés de l'aide alimentaire entre 2013 et 2016</h3>
</div>

In [ ]:
#Création d'un dataframe avec la zone, l'année et l'aide alimentaire puis groupby sur zone et année 
# Filtrer les données pour les années 2013 à 2016
aideAlimentaire_2013_2016 = aideAlimentaire[(aideAlimentaire['Année'] >= 2013) & (aideAlimentaire['Année'] <= 2016)]

# Grouper par pays et année, puis calculer le total de l'aide alimentaire
aideAlimentaire_grouped = aideAlimentaire_2013_2016.groupby(['Zone', 'Année'])['Valeur'].sum().reset_index()
print('Aide alimentaire groupée par pays et année : ','\n', aideAlimentaire_grouped)


Aide alimentaire groupée par pays et année :  
             Zone  Année     Valeur
0    Afghanistan   2013  128238000
1    Afghanistan   2014   57214000
2        Algérie   2013   35234000
3        Algérie   2014   18980000
4        Algérie   2015   17424000
..           ...    ...        ...
223       Égypte   2013    1122000
224     Équateur   2013    1362000
225     Éthiopie   2013  591404000
226     Éthiopie   2014  586624000
227     Éthiopie   2015  203266000

[228 rows x 3 columns]


In [ ]:
#Création d'une liste contenant les 5 pays qui ont le plus bénéficiées de l'aide alimentaire
top_5_pays_par_annee = aideAlimentaire_grouped.groupby('Année').apply(lambda x: x.nlargest(5, 'Valeur')).reset_index(drop=True)
print(top_5_pays_par_annee)

                         Zone  Année     Valeur
0                    Éthiopie   2013  591404000
1   République arabe syrienne   2013  563566000
2                      Soudan   2013  330230000
3                       Yémen   2013  264764000
4                       Kenya   2013  220966000
5   République arabe syrienne   2014  651870000
6                    Éthiopie   2014  586624000
7               Soudan du Sud   2014  450610000
8                      Soudan   2014  321904000
9                       Kenya   2014  217418000
10  République arabe syrienne   2015  524949000
11                      Yémen   2015  372306000
12                   Éthiopie   2015  203266000
13                      Kenya   2015  114452000
14                      Tchad   2015   73678000
15                      Yémen   2016  465574000
16  République arabe syrienne   2016  118558000
17                      Niger   2016   92742000
18                      Haïti   2016   12462000
19                    Algérie   2016    

/var/folders/fk/2614w5d17vl_s_3w8_y9cvkr0000gn/T/ipykernel_38343/2564069730.py:2: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  top_5_pays_par_annee = aideAlimentaire_grouped.groupby('Année').apply(lambda x: x.nlargest(5, 'Valeur')).reset_index(drop=True)


In [ ]:
#On filtre sur le dataframe avec notre liste
# Tri des pays par le total de l'aide alimentaire reçue sur la période 2013-2016
top_pays_aide_alimentaire = aideAlimentaire_grouped.groupby('Zone')['Valeur'].sum().nlargest(5).index

# Filtrer les données pour inclure uniquement les 5 pays sélectionnés
top_pays_aide_alimentaire_data = aideAlimentaire_grouped[aideAlimentaire_grouped['Zone'].isin(top_pays_aide_alimentaire)]

# Affichage des données
print(top_pays_aide_alimentaire_data)



                          Zone  Année     Valeur
157  République arabe syrienne   2013  563566000
158  République arabe syrienne   2014  651870000
159  République arabe syrienne   2015  524949000
160  République arabe syrienne   2016  118558000
189                     Soudan   2013  330230000
190                     Soudan   2014  321904000
191                     Soudan   2015   17650000
192              Soudan du Sud   2013  196330000
193              Soudan du Sud   2014  450610000
194              Soudan du Sud   2015   48308000
214                      Yémen   2013  264764000
215                      Yémen   2014  103840000
216                      Yémen   2015  372306000
217                      Yémen   2016  465574000
225                   Éthiopie   2013  591404000
226                   Éthiopie   2014  586624000
227                   Éthiopie   2015  203266000


In [ ]:
# Affichage des pays avec l'aide alimentaire par année
# Grouper par année et pays, puis calculer la somme de l'aide alimentaire pour chaque année
aide_alimentaire_par_annee = aideAlimentaire.groupby(['Zone', 'Année'])['Valeur'].sum().reset_index()

# Afficher les données
print(aide_alimentaire_par_annee)



            Zone  Année     Valeur
0    Afghanistan   2013  128238000
1    Afghanistan   2014   57214000
2        Algérie   2013   35234000
3        Algérie   2014   18980000
4        Algérie   2015   17424000
..           ...    ...        ...
223       Égypte   2013    1122000
224     Équateur   2013    1362000
225     Éthiopie   2013  591404000
226     Éthiopie   2014  586624000
227     Éthiopie   2015  203266000

[228 rows x 3 columns]


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.9 - Pays avec le moins de disponibilité par habitant</h3>
</div>

In [ ]:
#Calcul de la disponibilité en kcal par personne par jour par pays
# Calculer la disponibilité alimentaire en kcal par habitant pour chaque produit
dispo_alim_kcal_par_habitant = dispoAlimentaire.groupby(['Zone', 'Produit'])['Disponibilité alimentaire (Kcal/personne/jour)'].sum().reset_index()
print('Disponibilité alimentaire en kcal par habitant pour chaque produit : ','\n', dispo_alim_kcal_par_habitant)


Disponibilité alimentaire en kcal par habitant pour chaque produit :  
                Zone                Produit  \
0       Afghanistan       Abats Comestible   
1       Afghanistan        Agrumes, Autres   
2       Afghanistan  Aliments pour enfants   
3       Afghanistan                 Ananas   
4       Afghanistan                Bananes   
...             ...                    ...   
15411  Îles Salomon       Viande de Suides   
15412  Îles Salomon    Viande de Volailles   
15413  Îles Salomon          Viande, Autre   
15414  Îles Salomon                    Vin   
15415  Îles Salomon         Épices, Autres   

       Disponibilité alimentaire (Kcal/personne/jour)  
0                                                 5.0  
1                                                 1.0  
2                                                 1.0  
3                                                 0.0  
4                                                 4.0  
...                                    

In [ ]:
#Affichage des 10 pays qui ont le moins de dispo alimentaire par personne 
pays_min_dispo_alim = dispo_alim_kcal_par_habitant.groupby('Zone')['Disponibilité alimentaire (Kcal/personne/jour)'].sum().nsmallest(10)
print('Les 10 pays avec la disponibilité alimentaire la plus faible par habitant : ','\n', pays_min_dispo_alim)


Les 10 pays avec la disponibilité alimentaire la plus faible par habitant :  
 Zone
République centrafricaine                     1879.0
Zambie                                        1924.0
Madagascar                                    2056.0
Afghanistan                                   2087.0
Haïti                                         2089.0
République populaire démocratique de Corée    2093.0
Tchad                                         2109.0
Zimbabwe                                      2113.0
Ouganda                                       2126.0
Timor-Leste                                   2129.0
Name: Disponibilité alimentaire (Kcal/personne/jour), dtype: float64


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.10 - Pays avec le plus de disponibilité par habitant</h3>
</div>

In [ ]:
#Affichage des 10 pays qui ont le plus de dispo alimentaire par personne 
dispo_alim_kcal_par_habitant = dispoAlimentaire.groupby(['Zone', 'Produit'])['Disponibilité alimentaire (Kcal/personne/jour)'].sum().reset_index()

pays_max_dispo_alim = dispo_alim_kcal_par_habitant.groupby('Zone')['Disponibilité alimentaire (Kcal/personne/jour)'].sum().nlargest(10)
print('Les 10 pays avec la disponibilité alimentaire la plus élevée par habitant : ','\n', pays_max_dispo_alim)


Les 10 pays avec la disponibilité alimentaire la plus élevée par habitant :  
 Zone
Autriche                 3770.0
Belgique                 3737.0
Turquie                  3708.0
États-Unis d'Amérique    3682.0
Israël                   3610.0
Irlande                  3602.0
Italie                   3578.0
Luxembourg               3540.0
Égypte                   3518.0
Allemagne                3503.0
Name: Disponibilité alimentaire (Kcal/personne/jour), dtype: float64


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.11 - Exemple de la Thaïlande pour le Manioc</h3>
</div>

In [ ]:
#création d'un dataframe avec uniquement la Thaïlande 
# Créer un DataFrame pour la Thaïlande
thailande = dispoAlimentaire[dispoAlimentaire['Zone'] == 'Thaïlande'].copy()
print(thailande)


            Zone                 Produit   Origine  Aliments pour animaux  \
13570  Thaïlande        Abats Comestible   animale                    0.0   
13571  Thaïlande         Agrumes, Autres  vegetale                    0.0   
13572  Thaïlande  Alcool, non Comestible  vegetale                    0.0   
13573  Thaïlande   Aliments pour enfants  vegetale                    0.0   
13574  Thaïlande                  Ananas  vegetale                    0.0   
...          ...                     ...       ...                    ...   
13660  Thaïlande        Viande de Suides   animale                    0.0   
13661  Thaïlande     Viande de Volailles   animale                    0.0   
13662  Thaïlande           Viande, Autre   animale                    0.0   
13663  Thaïlande                     Vin  vegetale                    0.0   
13664  Thaïlande          Épices, Autres  vegetale                    0.0   

       Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour) 

In [ ]:
#Calcul de la sous nutrition en Thaïlande
sous_nutrition_thailande = sousNutrition[sousNutrition['Zone'] == 'Thaïlande']
print(sous_nutrition_thailande)

           Zone  Année  sous_nutrition
3330  Thaïlande   2012         6200000
3331  Thaïlande   2013         6200000
3332  Thaïlande   2014         6200000
3333  Thaïlande   2013         6000000
3334  Thaïlande   2014         6000000
3335  Thaïlande   2015         6000000
3336  Thaïlande   2014         5900000
3337  Thaïlande   2015         5900000
3338  Thaïlande   2016         5900000
3339  Thaïlande   2015         6000000
3340  Thaïlande   2016         6000000
3341  Thaïlande   2017         6000000
3342  Thaïlande   2016         6200000
3343  Thaïlande   2017         6200000
3344  Thaïlande   2018         6200000
3345  Thaïlande   2017         6500000
3346  Thaïlande   2018         6500000
3347  Thaïlande   2019         6500000


In [ ]:
## On calcule la proportion exportée en fonction de la proportion
# disponible intérieurement
# Créer une colonne 'proportion_exportee' dans le DataFrame thailande
thailande.loc[:, 'proportion_exportee'] = thailande['Exportations - Quantité'] / thailande['Disponibilité intérieure'] * 100
#En utilisant .loc[], on garanie qu'on modifie directement le DataFrame thailande et non une vue sur celui-ci
print(thailande)

            Zone                 Produit   Origine  Aliments pour animaux  \
13570  Thaïlande        Abats Comestible   animale                    0.0   
13571  Thaïlande         Agrumes, Autres  vegetale                    0.0   
13572  Thaïlande  Alcool, non Comestible  vegetale                    0.0   
13573  Thaïlande   Aliments pour enfants  vegetale                    0.0   
13574  Thaïlande                  Ananas  vegetale                    0.0   
...          ...                     ...       ...                    ...   
13660  Thaïlande        Viande de Suides   animale                    0.0   
13661  Thaïlande     Viande de Volailles   animale                    0.0   
13662  Thaïlande           Viande, Autre   animale                    0.0   
13663  Thaïlande                     Vin  vegetale                    0.0   
13664  Thaïlande          Épices, Autres  vegetale                    0.0   

       Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour) 

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 6 - Analyse complémentaires</h2>
</div>

In [ ]:
#Rajouter en dessous toutes les analyses complémetaires suite à la demande de mélanie :
#"et toutes les infos que tu trouverais utiles pour mettre en relief les pays qui semblent être 
#le plus en difficulté au niveau alimentaire"
#Calcul de la proportion de personnes sous alimentées par pays en 2017




Les zones qui importent plus de céréales qu'elles n'en exportent sont : 
                 Zone      Solde
7         Afghanistan -1173000.0
32        Afghanistan    -1000.0
40        Afghanistan   -10000.0
47        Afghanistan  -119000.0
67     Afrique du Sud   -21000.0
...               ...        ...
15321        Éthiopie   -89000.0
15324        Éthiopie   -18000.0
15348    Îles Salomon   -14000.0
15386    Îles Salomon    -1000.0
15402    Îles Salomon   -47000.0

[835 rows x 2 columns]


/var/folders/fk/2614w5d17vl_s_3w8_y9cvkr0000gn/T/ipykernel_12904/6356121.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cereales['Solde'] = cereales['Exportations - Quantité'] - cereales['Importations - Quantité']
